controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/InternalStepRunner.java

```diff
@@ -525,38 +542,58 @@ private String humanize(Node.ServiceState state) {
             return Optional.of(error);
         }
         logEndpoints(endpoints, logger);
 
         Optional<URI> testerEndpoint = controller.jobController().testerEndpoint(id);
-        if (testerEndpoint.isEmpty()) {
-            logger.log(WARNING, "Endpoints for the tester container vanished again, while it was still active!");
-            return Optional.of(error);
-        }
+        if (useConfigServerForTesterAPI(zoneId)) {
+            if ( ! controller.serviceRegistry().configServer().isTesterReady(getTesterDeploymentId(id))) {
+                logger.log(WARNING, "Tester container went bad!");
+                return Optional.of(error);
+            }
+        } else {
+            if (testerEndpoint.isEmpty()) {
+                logger.log(WARNING, "Endpoints for the tester container vanished again, while it was still active!");
+                return Optional.of(error);
+            }
 
-        if ( ! controller.jobController().cloud().testerReady(testerEndpoint.get())) {
-            logger.log(WARNING, "Tester container went bad!");
-            return Optional.of(error);
+            if ( ! controller.jobController().cloud().testerReady(testerEndpoint.get())) {
+                logger.log(WARNING, "Tester container went bad!");
+                return Optional.of(error);
+            }
         }
 
         logger.log("Starting tests ...");
-        controller.jobController().cloud().startTests(testerEndpoint.get(),
-                                                      TesterCloud.Suite.of(id.type(), isSetup),
-                                                      testConfigSerializer.configJson(id.application(),
-                                                                                      id.type(),
-                                                                                      true,
-                                                                                      endpoints,
-                                                                                      controller.applications().contentClustersByZone(deployments)));
+        TesterCloud.Suite suite = TesterCloud.Suite.of(id.type(), isSetup);
+        byte[] config = testConfigSerializer.configJson(id.application(),
+                                                        id.type(),
+                                                        true,
+                                                        endpoints,
+                                                        controller.applications().contentClustersByZone(deployments));
+        if (useConfigServerForTesterAPI(zoneId)) {
+            controller.serviceRegistry().configServer().startTests(getTesterDeploymentId(id), suite, config);
+        } else {
+            controller.jobController().cloud().startTests(testerEndpoint.get(), suite, config);
+        }
         return Optional.of(running);
     }
 
+    private boolean testerReady(RunId id, URI testerEndpoint) {
```

Comment:
Was this used? (Reviewing on phone...)

Reply:
Oh, forgot to remove that one (ended up not being able to use it after all), will remove in next PR